### PR TITLE
Issue #3130311 by SV: Override default view mode for activity stream views block if it was added via layout builde

### PR DESCRIPTION
**Pull request by volodymyr-sydor-lemberg**
## Problem
Activity stream views block which was added via layout builder should have a specific view mode

## Solution

- Add setting field on the form block (like "block title" or "items per page")
- Override view mode if this option is enabled

## Issue tracker

- https://getopensocial.atlassian.net/browse/YANG-2578
- https://www.drupal.org/project/social/issues/3130311

## How to test
- [ ] Create new dashboard node
- [ ] Add activity stream and save updates
- [ ] Dashboard node should be styled specifically (items in the list should use featured view mode)

### CODE REVIEW

File: themes/socialbase/templates/activity/activity--featured--sky.html.twig [hunk 1]
```diff
@@ -0,0 +1 @@
+{% extends "activity.html.twig" %}
```
Comment (ribel) [inline themes/socialbase/templates/activity/activity--featured--sky.html.twig:1]: @volodymyr-sydor-lemberg Is it really needed to have this template? It looks for me that we can remove it.

Comment (volodymyr-sydor-lemberg) [reply themes/socialbase/templates/activity/activity--featured--sky.html.twig:1]: @ribel looks like currently, we don't need it, i will remove it and update related PR

Comment (volodymyr-sydor-lemberg) [reply themes/socialbase/templates/activity/activity--featured--sky.html.twig]: done

**Comment by ronaldtebrake** [comment]
Screen Shot 2020-05-04 at 14 04 38

Posts and comments on posts still seem to make the stream on the dashboard look off. They probably need to have an updated featured view mode as well.

File: modules/social_features/social_activity/social_activity.install [hunk 2]
```diff
@@ -167,3 +171,72 @@ function social_activity_update_8801() {
   $view->set('display', $displays);
   $view->save(TRUE);
 }
+
+/**
+ * Create "Featured" view mode/display for activity.
+ */
+function social_activity_update_8802() {
+  // Create a new profile entity view mode.
```
Comment (ribel) [inline modules/social_features/social_activity/social_activity.install:179]: @volodymyr-sydor-lemberg please fix comment here.

File: modules/social_features/social_activity/social_activity.install [hunk 3]
```diff
@@ -167,3 +171,72 @@ function social_activity_update_8801() {
...
+  EntityViewDisplay::create($display)->save();
+}
+
+/**
+ * Create "Featured" view mode/display for post.
+ *
+ * @todo: should be moved into social_post module.
+ */
+function social_activity_update_8803() {
+  // Create a new profile entity view mode.
```
Comment (ribel) [inline modules/social_features/social_activity/social_activity.install:207]: @volodymyr-sydor-lemberg please fix comment here.

File: modules/social_features/social_post/config/install/core.entity_view_display.post.photo.featured.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - field.field.post.photo.field_post_comments
+    - field.field.post.photo.field_post_image
+    - field.field.post.photo.field_recipient_group
+    - field.field.post.photo.field_recipient_user
+    - field.field.post.photo.field_visibility
+    - image.style.social_post_photo
+    - social_post.post_type.photo
+  module:
+    - image
+    - lazy
```
Comment (ribel) [inline modules/social_features/social_post/config/install/core.entity_view_display.post.photo.featured.yml:16]: @volodymyr-sydor-lemberg This configuration objects have unmet dependencies in Travis log - please check and fix. Maybe lazy image not needed here?

File: modules/social_features/social_post/templates/post--featured.html.twig [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+              </a>
+            {% endif %}
+          </div>
+        </div>
+      {% endblock %}
+
+      {% block comments %}
+        {% if logged_in %}
+          <div class="card__nested-section">
+            {{ content.field_yammer_comments }}
```
Comment (ribel) [inline modules/social_features/social_post/templates/post--featured.html.twig:70]: @volodymyr-sydor-lemberg Please remove all yammer specific things from Distro PR and add them only in the project.

Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_post/templates/post--featured.html.twig]: this one already should be done

File: modules/social_features/social_activity/social_activity.install (hunk 3 above)
Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_activity/social_activity.install]: done

File: modules/social_features/social_activity/social_activity.install (hunk 2 above)
Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_activity/social_activity.install]: done

**Comment by ribel** [comment]
Hi @volodymyr-sydor-lemberg Is work on this PR finished? 
And can we merge it and include this in 8.4 release without #1783?

**Comment by volodymyr-sydor-lemberg** [comment]
@ribel Yes, PR is ready for review and it's already merged into related one - #1783, so we could include it into the release

**Comment by ribel** [comment]
🍒cherry-picked to 8.x-8.x

**Pull request merged**